### PR TITLE
Make `mlflow ui` an alias for `mlflow server`

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## Related Issues/PRs

 #7139

## What changes are proposed in this pull request?

Make `mlflow ui` an alias for `mlflow server`.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
Test

```
> mlflow ui --help
Usage: mlflow server [OPTIONS]

  Run the MLflow tracking server.

  The server listens on http://localhost:5000 by default and only accepts
  connections from the local machine. To let the server accept connections
  from other machines, you will need to pass ``--host 0.0.0.0`` to listen on
  all network interfaces (or a specific interface address).

Options:
  --backend-store-uri PATH     URI to which to persist experiment and run
                               data. Acceptable URIs are SQLAlchemy-compatible
                               database connection strings (e.g.
                               'sqlite:///path/to/file.db') or local
                               filesystem URIs (e.g.
                               'file:///absolute/path/to/directory'). By
                               default, data will be logged to the ./mlruns
                               directory.
  --registry-store-uri URI     URI to which to persist registered models.
                               Acceptable URIs are SQLAlchemy-compatible
                               database connection strings (e.g.
                               'sqlite:///path/to/file.db'). If not specified,
                               `backend-store-uri` is used.
  --default-artifact-root URI  Directory in which to store artifacts for any
                               new experiments created. For tracking server
                               backends that rely on SQL, this option is
                               required in order to store artifacts. Note that
                               this flag does not impact already-created
                               experiments with any previous configuration of
                               an MLflow server instance. By default, data
                               will be logged to the mlflow-artifacts:/ uri
                               proxy if the --serve-artifacts option is
                               enabled. Otherwise, the default location will
                               be ./mlruns.
  --serve-artifacts            If specified, enables serving of artifact
                               uploads, downloads, and list requests by
                               routing these requests to the storage location
                               that is specified by '--artifact-destination'
                               directly through a proxy. The default location
                               that these requests are served from is a local
                               './mlartifacts' directory which can be
                               overridden via the '--artifacts-destination'
                               argument. Default: False
  --artifacts-only             If specified, configures the mlflow server to
                               be used only for proxied artifact serving. With
                               this mode enabled, functionality of the mlflow
                               tracking service (e.g. run creation, metric
                               logging, and parameter logging) is disabled.
                               The server will only expose endpoints for
                               uploading, downloading, and listing artifacts.
                               Default: False
  --artifacts-destination URI  The base artifact location from which to
                               resolve artifact upload/download/list requests
                               (e.g. 's3://my-bucket'). Defaults to a local
                               './mlartifacts' directory. This option only
                               applies when the tracking server is configured
                               to stream artifacts and the experiment's
                               artifact root location is http or mlflow-
                               artifacts URI.
  -h, --host HOST              The network address to listen on (default:
                               127.0.0.1). Use 0.0.0.0 to bind to all
                               addresses if you want to access the tracking
                               server from other machines.
  -p, --port INTEGER           The port to listen on (default: 5000).
  -w, --workers TEXT           Number of gunicorn worker processes to handle
                               requests (default: 4).
  --static-prefix TEXT         A prefix which will be prepended to the path of
                               all static paths.
  --gunicorn-opts TEXT         Additional command line options forwarded to
                               gunicorn processes.
  --waitress-opts TEXT         Additional command line options for waitress-
                               serve.
  --expose-prometheus TEXT     Path to the directory where metrics will be
                               stored. If the directory doesn't exist, it will
                               be created. Activate prometheus exporter to
                               expose metrics on /metrics endpoint.
  --help                       Show this message and exit.
```

File: mlflow/cli.py [hunk 1]
```diff
@@ -34,7 +34,18 @@
...
-@click.group()
+class AliasedGroup(click.Group):
+    def get_command(self, ctx, cmd_name):
+        # `mlflow ui` is an alias for `mlflow server`
+        cmd_name = "server" if cmd_name == "ui" else cmd_name
+        return super().get_command(ctx, cmd_name)
+
+    def resolve_command(self, ctx, args):
+        _, cmd, args = super().resolve_command(ctx, args)
+        return cmd.name, cmd, args
```
Comment (harupy) [inline mlflow/cli.py:45]: Ref: https://click.palletsprojects.com/en/8.1.x/advanced/#command-aliases

Comment (harupy) [reply mlflow/cli.py]: With 

```bash
# With resolve_command
> mlflow ui --help
Usage: mlflow server [OPTIONS]

# Without resolve_command
> mlflow ui --help
Usage: mlflow ui [OPTIONS]
```

Comment (harupy) [reply mlflow/cli.py]: Removed `resolve_command`.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !

**Pull request merged**